lock-api/src/main/java/com/palantir/lock/watch/NoOpLockWatchEventCache.java

```diff
@@ -58,11 +67,34 @@ public TransactionsLockWatchUpdate getUpdateForTransactions(
             Set<Long> startTimestamps, Optional<LockWatchVersion> version) {
         return ImmutableTransactionsLockWatchUpdate.builder()
                 .clearCache(true)
-                .startTsToSequence(
-                        startTimestamps.stream().collect(Collectors.toMap(startTs -> startTs, $ -> FAKE_VERSION)))
+                .startTsToSequence(startTimestamps.stream()
+                        .collect(Collectors.toMap(startTs -> startTs, $ -> currentVersion.orElse(FAKE_VERSION))))
                 .build();
     }
 
     @Override
     public void removeTransactionStateFromCache(long startTimestamp) {}
+
+    private void updateVersion(Optional<LockWatchVersion> maybeNewVersion) {
+        currentVersion = maybeNewVersion.map(newVersion -> currentVersion
+                .filter(current -> current.id().equals(newVersion.id()) && current.version() > newVersion.version())
+                .orElse(newVersion));
```

Comment:
from discussion below: can we use imperative statements here? I find this difficult to parse (I read the filter as a top-level thing that rejects versions that were unfriendly, when that's not the case).